scanpy/metrics/_morans_i.py

```diff
@@ -0,0 +1,247 @@
+"""Moran's I global spatial autocorrelation."""
+from typing import Union, Optional
+from functools import singledispatch
+from anndata import AnnData
+
+import numpy as np
+from scipy import sparse
+from numba import njit, prange
+
+from scanpy.get import _get_obs_rep
+from scanpy.metrics._gearys_c import _resolve_vals
+
+
+@singledispatch
+def morans_i(
+    adata: AnnData,
+    *,
+    vals: Optional[Union[np.ndarray, sparse.spmatrix]] = None,
+    use_graph: Optional[str] = None,
+    layer: Optional[str] = None,
+    obsm: Optional[str] = None,
+    obsp: Optional[str] = None,
+    use_raw: bool = False,
+) -> Union[np.ndarray, float]:
+    r"""
+    Calculate Moran’s I Global Autocorrelation Statistic.
+
+    Moran’s I is a global autocorrelation statistic for some measure on a graph. It is commonly used in
+    spatial data analysis to assess autocorrelation on a 2D grid. It is closely related to Geary's C,
+    but not identical. More info can be found `here`<https://en.wikipedia.org/wiki/Moran%27s_I> .
+
+    .. math::
+
+        I=\frac{n}{S_{0}} \frac{\sum_{i=1}^{n} \sum_{j=1}^{n} w_{i, j} z_{i} z_{j}}{\sum_{i=1}^{n} z_{i}^{2}}
```

Comment:
```suggestion
    .. math::

        I =
            \frac{
                N \sum_{i, j} w_{i, j} z_{i} z_{j}
            }{
                S_{0} \sum_{i} z_{i}^{2}
            } 
```

Suggestion for more consistency with gearys c.

Could also be nice to define the terms more clearly, but that would the case for gearys c as well. Not needed, just a thought.